[feat] template setting, 데이터 렌더링 및 필터링 기능

migugin opened the pull request:
### 6/11 Daily
- Vue2 프로젝트 템플릿 구성 및 필요 라이브러리 설치
- SearchBox, PostList 마크업
- 기능 구현
    - 데이터 렌더링 기능
    - 텍스트 필터링 기능
- 상태 관리를 위한 Store 구성 (Vuex)

migugin committed:
[feat] DB 데이터 리스트 렌더링 기능

- Vuex 사용하여 상태 관리
- 라우터 상에 Feed 페이지 root path 지정

migugin committed:
[feat] 검색 폼 구현, 텍스트 필터링 기능

migugin committed:
[feat] 클릭 이벤트 전달 방식 변경

MANAPIE commented on src/components/AppInput.vue:
```diff
@@ -0,0 +1,24 @@
+  <div>
+    <label :for="id">
+      {{ label }}
+    </label>
```
label 값이 없으면 라벨을 아예 안 보여주는 게 낫지 않을지

migugin replied:
`hiddenLabel` props 추가해서 label이 UI에 표시되지 않아야 하는 label은 숨길 수 있도록 기능 추가했습니다

migugin replied:
`hiddenLabel` props 추가해서 UI에 표시되지 않아야 하는 `label` 요소는 숨길 수 있도록 기능 추가했습니다

MANAPIE replied:
아예 없게 해주세요
내용 없는 라벨은 혼란스럽기만 하고 쓸모가 없어요

MANAPIE replied:
label 값이 있는지만 보면 되는데 hiddenLabel 프롭을 또 추가한 이유가 있을까요?

migugin replied:
접근성 관련된 내용인데 `input` 요소가 무슨 용도이고 뭘 입력해야 하는지 스크린 리더 사용자가 식별할 수 있어야 해서 `label` 태그를 붙여줍니다
`label` 태그 없이 `input`만 사용하면 `<input type="date" />` 읽어줄 때 '날짜 입력 요소' 정도로 알려주고 입력해야 하는 날짜가 시작 날짜인지, 종료 날짜인지 알려주지 않습니다

hiddenLabel 프로퍼티 사용한 이유는 `label` 태그를 사용했지만 UI에 표시되지 않아야 할 때 스크린리더가 인식할 수 있게끔 감추는 방법(accessibility hidden)을 권장하고 있어서 이렇게 사용하였습니다

MANAPIE replied:
어.. 저도 그 부분 고려하고 있긴 한데..
제 말은 label 안에 텍스트가 없으면 스크린리더가 어떻게 읽어줘요..?
라벨. 텍스트 입력 요소. 이렇게만 읽어주는 게 무슨 의미가 있나 해서요
그럴 바에야 라벨이라도 빼서 혼란을 줄이려고 한 거예요

migugin replied:
아아아 label 안에 텍스트 `label: String` 프로퍼티로 받아서 출력해줍니다!!!
```html
<label :for="id" :class="labelClass">
  {{ label }}
</label>
```
이거는 제가 labelText로 프로퍼티 이름을 바꾸던가 했어야 했네요

MANAPIE replied:
접근성 이슈로 label이 필요하다는 건 알겠는데요
이걸 required: true로 강제할 게 아니면
없을 수도 있다는 거잖아요

1. 없을 수도 있을 걸 고려해서 짜기
2. required 넣어서 없으면 안 되게 하기

여기 말고도 프롭 전반적으로 체크하셔서 필수 요소들에 대해서는 required 지정해줘야겠네요

MANAPIE replied:
추가로.. 디자인 상 label이 제공되기 어려운 경우에는 아래 방법으로 레이블링을 할 수 있으니 참고하세요

1. label의 스타일을 숨김 처리 (display none은 스크린리더도 안 읽으므로 주의)
2. label 요소 대신 title 어트리뷰트 사용하기

migugin replied:
required 지정해주는 부분은 생각 못했습니다
필수인 요소들은 이거 다 추가해주는 게 맞는 것 같네요
내일 같이 작업하겠습니다!

MANAPIE commented on src/components/AppInput.vue:
```diff
@@ -0,0 +1,24 @@
+    label: String,
+    id: String,
+    placeholder: String,
+    role: String,
```
role은 이 컴포넌트에서 안 쓰이고 있네요
<input>에 넣으려는 거였을까요?
그리고 기본값 지정 필요해보입니다.

migugin replied:
컴포넌트에서 누락된 것이 맞습니다!
props 추가하고 기본값 지정하였습니다

MANAPIE commented on src/components/FeedView/FeedItem.vue:
```diff
@@ -0,0 +1,16 @@
+  <li>
+    <h3>{{ postData.title }}</h3>
+    <p>{{ postData.date }}</p>
+    <p>{{ postData.content }}</p>
```
콘텐츠에 줄바꿈 등 들어가지 않나요
대응이 안 돼 보이네요

migugin replied:
width를 넘기는 텍스트는 개행될 수 있도록 CSS 추가했습니다.
```CSS
p {
  overflow-wrap: break-word;
  white-space: pre-line;
}
```

MANAPIE commented on src/components/AppButton.vue:
```diff
@@ -0,0 +1,23 @@
+<template>
+  <button :type="type" :class="['button', className]" @click="handleClickEvent">
```
가독성 향상을 위해 한 줄에 한 어트리뷰트만 쓰는 것을 권장합니다

migugin replied:
prettier 사용해서 포맷팅 된 부분인데 printWidth 낮춰서 속성 3개 이상일 때 개행할 수 있도록 하겠습니다~

MANAPIE commented on src/components/AppInput.vue:
```diff
@@ -0,0 +1,24 @@
+    <label :for="id">
+      {{ label }}
+    </label>
+    <input :id="id" :type="type" :placeholder="placeholder" ref="inputRef"/>
```
가독성 향상을 위해 한 줄에 한 어트리뷰트만 쓰는 것을 권장합니다

migugin replied:
✅

MANAPIE commented on src/components/FeedView/FeedList.vue:
```diff
@@ -0,0 +1,24 @@
+<template>
+  <ul>
+    <FeedItem v-for="data in renderData" :key="data.id" :postData="data" />
```
가독성 향상을 위해 한 줄에 한 어트리뷰트만 쓰는 것을 권장합니다

migugin replied:
✅

MANAPIE commented on src/components/FeedView/SearchBox.vue:
```diff
@@ -0,0 +1,60 @@
+    <AppInput
+      :id="'search-title'"
+      :label="'제목 검색'"
+      role="searchbox"
```
얘만 : 안 붙은 게 맞을까요
통일시킵시다

migugin replied:
role, placeholder, ref 모두 수정했습니다

MANAPIE commented on src/components/FeedView/SearchBox.vue:
```diff
@@ -0,0 +1,60 @@
+      :id="'search-title'"
+      :label="'제목 검색'"
+      role="searchbox"
+      placeholder="검색어를 입력하새요."
```
똑같이 스트링으로 들어가고 프롭으로 넘어가는 건데
어디는 : 붙어있고 어디는 안 붙어있고 그러네요
통일시키는 게 좋을 듯!

migugin replied:
✅

migugin committed:
[feat] PR 피드백 반영

- label 요소 숨김 기능 추가
- attribute 개행 및 스타일 수정
- text content 줄바꿈 대응
- 누락 props 추가

MANAPIE commented on src/components/AppInput.vue:
```diff
@@ -0,0 +1,40 @@
+<template>
+  <div>
+    <label :for="id" :class="hiddenLabel && 'a11y-hidden'">
```
프리티어 썼다고 다 믿지 말고 눈으로도 확인 해주세요

migugin replied:
어트리뷰트 3개 이상 사용되어 한눈에 파악하기 힘들 때 개행하도록 한다고 피드백 남겨서 일단 코드를 바꾸진 않았습니다!
오늘 작업한 내용 중에 이 부분 프로퍼티가 추가되어서 개행해놨습니당

MANAPIE replied:
개행할 거면 다 하는 게 낫지 않을까 싶네요
3개 이상인 것을 거르겠다는 용도가 아니라 어트리뷰트를 명확히 구분해서 보겠다는 목적인데
1개만 있을 때랑 2개만 있을 때가 줄바꿈 차이가 없으면 일관성은 둘째치고 2개일 때는 구분이 어려울 수 있어서요
일단 이건 확인~

MANAPIE reviewed (approved):
prop에 required 처리해주는 건 내일 작업
